updating APIs, removing obsolete API usage, fixing build warnings

jiri-malec opened the pull request:
# Motivation
 - Fixes issue [`86198608`](https://github.com/orgs/Kentico/projects/14?pane=issue&itemId=86198608)
 - Removing obsolete API, upgrading API references, removing additional build warnings

## Checklist

- [ ] Code follows coding conventions held in this repo
- [ ] Automated tests have been added
- [ ] Tests are passing
- [ ] Docs have been updated (if applicable)
- [ ] Temporary settings (e.g. variables used during development and testing) have been reverted to defaults

## How to test

If manual testing is required, what are the steps?

Rumec reviewed (changes requested):
Look at my proposition, please.

If you have any questions, don't hesitate to ask!

seangwright commented:
Just a heads up, I did a lot of this work already in https://github.com/Kentico/xperience-by-kentico-zapier/pull/34

There is some project structured/config that was missing that I also added.

jiri-malec commented:
> Just a heads up, I did a lot of this work already in #34
> 
> There is some project structured/config that was missing that I also added.

Thank you for pointing that out! From my perspective, your merge request addresses all the issues I was targeting with this one. Therefore, it might make sense to close this MR and proceed with #34 instead. @Rumec

jiri-malec commented on examples/DancingGoat/Helpers/AreaRestrictionHelper.cs:
```diff
@@ -16,19 +12,16 @@ public static class AreaRestrictionHelper
 
             return GetWidgetsIdentifiers()
-                .Where(id => allowedScopes.Any(scope => id.StartsWith(scope, StringComparison.OrdinalIgnoreCase)))
+                .Where(id => allowedScopes.ToList().Exists(scope => id.StartsWith(scope, StringComparison.OrdinalIgnoreCase)))
```
Yes, good point. Since a HashSet has constant-time complexity for lookups, your solution would be more efficient.
